ostd/src/sync/mutex.rs

```diff
@@ -50,7 +50,11 @@ impl<T: ?Sized> Mutex<T> {
     pub fn try_lock(&self) -> Option<MutexGuard<T>> {
         // Cannot be reduced to `then_some`, or the possible dropping of the temporary
         // guard will cause an unexpected unlock.
-        self.acquire_lock().then_some(MutexGuard { mutex: self })
+        //
+        // TODO: Remove the lint suppression when the false positive warning is fixed
+        // upstream.
+        #[allow(clippy::unnecessary_lazy_evaluations)]
```

Comment:
@tatetian This TODO comment refers to https://github.com/asterinas/asterinas/commit/3790efb0d9c8c23b268ab3926910f90acb6ab9d1, which is addressed by https://github.com/asterinas/asterinas/commit/576578baf4025686ae4c2893c2aafbc8d1e14722. Would it be better to create a `new` method?

Reply:
I'll vote for it. If you do create a `new` method please mark it unsafe.

Reply:
OK, pls review.